medium-tests/src/test/java/mediumtest/synchronization/ServerInfoSynchronizationMediumTests.java

```diff
@@ -58,8 +58,43 @@ void it_should_pull_server_info_when_bound_configuration_scope_is_added() {
 
     waitAtMost(3, SECONDS).untilAsserted(() -> assertThat(getServerInfoFile())
       .exists()
-      .extracting(this::readServerVersion)
-      .isEqualTo("10.3"));
+      .extracting(this::readServerVersion, this::readServerMode)
+      .containsExactly("10.3", true));
+  }
+
+  @Test
+  void it_should_pull_server_info_and_mqr_mode_should_be_disabled() {
+    var server = newSonarQubeServer("10.1")
+      .withProject("projectKey", project -> project.withBranch("main"))
+      .start();
+    backend = newBackend()
+      .withEnabledLanguageInStandaloneMode(Language.JAVA)
+      .withSonarQubeConnection("connectionId", server)
+      .withFullSynchronization()
+      .build();
+
+    addConfigurationScope("configScopeId", "connectionId", "projectKey");
+
+    waitAtMost(3, SECONDS).untilAsserted(() -> assertThat(getServerInfoFile())
+      .exists()
+      .extracting(this::readServerVersion, this::readServerMode)
+      .containsExactly("10.1", false));
+  }
+
+  @Test
+  void it_should_pull_server_info_from_sonarcloud() {
+    backend = newBackend()
+      .withEnabledLanguageInStandaloneMode(Language.JAVA)
+      .withSonarCloudConnection("connectionId", "test")
+      .withFullSynchronization()
+      .build();
+
+    addConfigurationScope("configScopeId", "connectionId", "projectKey");
+
+    waitAtMost(3, SECONDS).untilAsserted(() -> assertThat(getServerInfoFile())
+      .exists()
+      .extracting(this::readServerMode)
+      .isEqualTo(true));
```

Comment:
Is this test realistic? Shouldn't we mimic the fact that this setting does not exist on SC and that the synchronization still works?